dc/s2n-quic-dc/src/event/generated.rs

```diff
@@ -0,0 +1,537 @@
+        fn endpoint_type(&self) -> &EndpointType;
+        #[doc = r" A context from which the event is being emitted"]
+        #[doc = r""]
+        #[doc = r" An event can occur in the context of an Endpoint or Connection"]
+        fn subject(&self) -> Subject;
+        #[doc = r" The time the event occurred"]
+        fn timestamp(&self) -> &crate::event::Timestamp;
+    }
+    impl Meta for ConnectionMeta {
+        fn endpoint_type(&self) -> &EndpointType {
+            &self.endpoint_type
+        }
+        fn subject(&self) -> Subject {
+            Subject::Connection { id: self.id }
+        }
+        fn timestamp(&self) -> &crate::event::Timestamp {
+            &self.timestamp
+        }
+    }
+    impl Meta for EndpointMeta {
+        fn endpoint_type(&self) -> &EndpointType {
+            &self.endpoint_type
+        }
+        fn subject(&self) -> Subject {
+            Subject::Endpoint {}
+        }
+        fn timestamp(&self) -> &crate::event::Timestamp {
+            &self.timestamp
+        }
+    }
+    #[doc = r" Allows for events to be subscribed to"]
+    pub trait Subscriber: 'static + Send {
+        #[doc = r" An application provided type associated with each connection."]
+        #[doc = r""]
+        #[doc = r" The context provides a mechanism for applications to provide a custom type"]
+        #[doc = r" and update it on each event, e.g. computing statistics. Each event"]
+        #[doc = r" invocation (e.g. [`Subscriber::on_packet_sent`]) also provides mutable"]
+        #[doc = r" access to the context `&mut ConnectionContext` and allows for updating the"]
+        #[doc = r" context."]
+        #[doc = r""]
+        #[doc = r" ```no_run"]
+        #[doc = r" # mod s2n_quic { pub mod provider { pub mod event {"]
+        #[doc = r" #     pub use s2n_quic_core::event::{api as events, api::ConnectionInfo, api::ConnectionMeta, Subscriber};"]
+        #[doc = r" # }}}"]
+        #[doc = r" use s2n_quic::provider::event::{"]
+        #[doc = r"     ConnectionInfo, ConnectionMeta, Subscriber, events::PacketSent"]
+        #[doc = r" };"]
+        #[doc = r""]
+        #[doc = r" pub struct MyEventSubscriber;"]
+        #[doc = r""]
+        #[doc = r" pub struct MyEventContext {"]
+        #[doc = r"     packet_sent: u64,"]
+        #[doc = r" }"]
+        #[doc = r""]
+        #[doc = r" impl Subscriber for MyEventSubscriber {"]
+        #[doc = r"     type ConnectionContext = MyEventContext;"]
+        #[doc = r""]
+        #[doc = r"     fn create_connection_context("]
+        #[doc = r"         &mut self, _meta: &ConnectionMeta,"]
+        #[doc = r"         _info: &ConnectionInfo,"]
+        #[doc = r"     ) -> Self::ConnectionContext {"]
+        #[doc = r"         MyEventContext { packet_sent: 0 }"]
+        #[doc = r"     }"]
+        #[doc = r""]
+        #[doc = r"     fn on_packet_sent("]
+        #[doc = r"         &mut self,"]
+        #[doc = r"         context: &mut Self::ConnectionContext,"]
```

Comment:
Hmm looks like we'll need to modify this example as well. But we can have that as a follow-up.

Reply:
Eh yeah, wasn't sure what we wanted to do with the docs. That will be a follow up.